platforms/documentation/docs/src/docs/userguide/releases/upgrading/upgrading_version_8.adoc

```diff
@@ -113,6 +113,12 @@ container.attributeProvider(firstAttribute, project.getProviders().provider(() -
 There is no good public use case for this exception, and it is not intended to be thrown by users.
 It will be replaced by `org.gradle.api.internal.artifacts.transform.VariantTransformConfigurationException` for internal use only in Gradle 9.0.
 
+[[deprecated_string_value_to_enum_rich_property_assignment]]
+
+Assigning a string value to an enum rich property e.g. `Property<T extends Enum>` is deprecated and will become an error with Gradle 10.0.
+This behaviour was implemented only as a compatibility feature for Groovy DSL for easier migration of core Gradle APIs to rich properties.
```

Comment:
(please check if this is syntactically correct)

```suggestion
Groovy supports link:https://groovy-lang.org/semantics.html#_string_to_enum_coercion[String to enum coercion]. Assigning a string to a property of type `Property<T>` where `T` is an enum is deprecated. This will become an error in Gradle 10.0.

This behavior is only applicable to plugins written in Groovy and the Groovy DSL.
```

Reply:
Thanks! Applied changes